### PR TITLE
drivers: sensor: stts751: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by benjaminbjornsson**

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

File: drivers/sensor/stts751/stts751.h [hunk 1]
```diff
@@ -20,21 +20,17 @@
 #include "stts751_reg.h"
 
 struct stts751_config {
-	char *master_dev_name;
+	struct i2c_dt_spec i2c;
```
Comment (MaureenHelm) [inline drivers/sensor/stts751/stts751.h:23]: I think this should be inside a conditional `#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)` like `i2c_slv_addr` was before this patch.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/stts751/stts751.h (hunk 1 above)
Comment (benjaminbjornsson) [reply drivers/sensor/stts751/stts751.h:23]: There's an `#error` if `#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)` is false. Looks to me as if the `#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)`'s could be removed all together?

Comment (benjaminbjornsson) [reply drivers/sensor/stts751/stts751.h:24]: I've added the conditional now!

File: drivers/sensor/stts751/stts751.c [hunk 2]
```diff
@@ -201,15 +200,14 @@ static int stts751_init(const struct device *dev)
 static struct stts751_data stts751_data;
 
 static const struct stts751_config stts751_config = {
-	.master_dev_name = DT_INST_BUS_LABEL(0),
+	.i2c = I2C_DT_SPEC_INST_GET(0),
```
Comment (gmarull) [inline drivers/sensor/stts751/stts751.c:203]: this is set unconditionally, `i2c` field will not always be available

Comment (benjaminbjornsson) [reply drivers/sensor/stts751/stts751.c]: Oh, thanks!

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**